src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -50,13 +57,35 @@ public static function register($mode = 0)
             if (false === $mode) {
                 $mode = getenv('SYMFONY_DEPRECATIONS_HELPER');
             }
-            if (DeprecationErrorHandler::MODE_WEAK !== $mode && (!isset($mode[0]) || '/' !== $mode[0])) {
+            if (DeprecationErrorHandler::MODE_WEAK !== $mode && DeprecationErrorHandler::MODE_WEAK_VENDORS !== $mode && (!isset($mode[0]) || '/' !== $mode[0])) {
                 $mode = preg_match('/^[1-9][0-9]*$/', $mode) ? (int) $mode : 0;
             }
 
             return $memoizedMode = $mode;
         };
 
+        $inVendors =  function ($path) {
+            if (null === self::$vendors) {
+                foreach (get_declared_classes() as $class) {
+                    if ('C' === $class[0] && 0 === strpos($class, 'ComposerAutoloaderInit')) {
+                        $r = new \ReflectionClass($class);
+                        $v = dirname(dirname($r->getFileName()));
+                        if (file_exists($v.'/composer/installed.json')) {
+                            self::$vendors[] = $v;
+                        }
+                    }
+                }
+            }
+            $path = realpath($path) ?: $path;
+            foreach (self::$vendors as $vendor) {
+                if (0 === strpos($path, $vendor)) {
```

Comment:
What is the second part about exactly? I was told it was about checking whether `$path` was a vendor dir itself, which wouldn't be useful in that case, would it?